### PR TITLE
Transfer roberta gnn

### PR DESCRIPTION
**Pull request by fvlieshout**
Implemented some extra things so we can do transfer learning for the Roberta GNN models

### CODE REVIEW

File: data_prep/roberta_graph_dataset.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def _generate_features(self):
                 docs = self._raw_texts[i * batch_size:max_docs]
                 encodings = self._tokenizer(docs, truncation=True, padding=True)
                 items = {k: torch.tensor(v, dtype=torch.long, device=self._device) for k, v in encodings.items()}
```
Comment (fvlieshout) [inline data_prep/roberta_graph_dataset.py]: I don't really know why you added the creation of this dictionary. I don't think we need it as my previous code worked fine but I guess you put this here for a reason so I left it in. I had to change the output of the doc_embedder though because we only need the final hidden states

File: data_prep/roberta_graph_dataset.py [hunk 2]
```diff
@@ -64,23 +64,14 @@ def get_doc_embedder(self):
...
-
         if self.roberta_checkpoint is None:
-            # use only pretrained roberta for the document embeddings
-            return encoder
-
-        # use pretrained + fine tuned roberta for the document embeddings
-        encoder.load_state_dict(self.get_encoder_state_dict())
-
-        return encoder
+            # use off-the-shelve pretrained Roberta Model
```
Comment (fvlieshout) [inline data_prep/roberta_graph_dataset.py:69]: We don't need to freeze the models as we are using torch.no_grad anyway whenever we use them for creating the document embeddings (that's the only place we use them). Now the code is much simpler

File: data_prep/roberta_graph_dataset.py (hunk 1 above)
Comment (rahelbeloch) [reply data_prep/roberta_graph_dataset.py]: Because we are using our `RobertaEncoder` model, with needs both, the `input_ids` and the `attention_mask`. Both is contained in the encodings returned by the tokenizer.

File: data_prep/roberta_graph_dataset.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def _generate_features(self):
                 docs = self._raw_texts[i * batch_size:max_docs]
                 encodings = self._tokenizer(docs, truncation=True, padding=True)
                 items = {k: torch.tensor(v, dtype=torch.long, device=self._device) for k, v in encodings.items()}
-                out, _ = doc_embedder(items)
+                _, out = doc_embedder(items['input_ids'])
```
Comment (rahelbeloch) [inline data_prep/roberta_graph_dataset.py]: Isn't this using the labels our `RobertaEncoder` returns? (see image)

Screenshot 2021-05-21 at 16 03 40

File: data_prep/roberta_graph_dataset.py (hunk 2 above)
Comment (rahelbeloch) [reply data_prep/roberta_graph_dataset.py:69]: jep

File: data_prep/roberta_graph_dataset.py [hunk 4]
```diff
@@ -64,23 +64,14 @@ def get_doc_embedder(self):
...
         if self.roberta_checkpoint is None:
-            # use only pretrained roberta for the document embeddings
-            return encoder
-
-        # use pretrained + fine tuned roberta for the document embeddings
-        encoder.load_state_dict(self.get_encoder_state_dict())
-
-        return encoder
+            # use off-the-shelve pretrained Roberta Model
+            embedder = RobertaModel.from_pretrained('roberta-base').to(self._device)
```
Comment (rahelbeloch) [inline data_prep/roberta_graph_dataset.py]: I think you can immetiately return this model here, as you are not doing any further operation on it...  then it becomes even more simple

File: data_prep/roberta_graph_dataset.py (hunk 3 above)
Comment (rahelbeloch) [reply data_prep/roberta_graph_dataset.py]: Ok, I see you are now not using our `RobertaEncoder` anymore. Then this is obsolete

File: data_prep/roberta_graph_dataset.py (hunk 1 above)
Comment (rahelbeloch) [reply data_prep/roberta_graph_dataset.py]: I see you are not using our `RobertaEncoder` anymore, then this dict creating can also be removed again.

**Review by rahelbeloch — COMMENTED** [review]
@fvlieshout made some comments.

File: data_prep/roberta_graph_dataset.py (hunk 3 above)
Comment (fvlieshout) [reply data_prep/roberta_graph_dataset.py]: No we still do use RobertaEncoder. But only if the path to that model is specified. I changed it now with an if statement

**Review by rahelbeloch — APPROVED** [review]
Looks good!
